packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -629,3 +636,23 @@ abstract class BindingBase {
 Future<void> _exitApplication() async {
   exit(0);
 }
+
+/// Additional configuration used for hot reload reassemble optimizations.
+///
+/// Do not extend, implement, or mixin this class. This may only be instantiated
+/// in debug mode.
+class DebugReassembleConfig {
+  /// Create a new [DebugReassembleConfig].
+  ///
+  /// Throws a [FlutterError] if this is called in profile or release mode.
+  DebugReassembleConfig({
+    this.widgetName,
+  }) {
+    if (!kDebugMode) {
+      throw FlutterError('Cannot instaniate DebugReassembleConfig in profile or release mode.');
+    }
+  }
+
+  /// The name of the widget that was modified, or `null` if the change was elsewhere.
+  final String? widgetName;
```

Comment:
This layer isn't supposed to know about widgets. Can we move this logic to the widgets layer?

Reply:
Looks like the only use at lower layers is to disable the rendering layer's reassemble logic? Maybe we can just skip calling the reassemble logic in the rendering layer and just make fastReassemble invoke the relevant bits (like resetting the error count) and directly reassemble the widget tree with the class name? I'll file a bug to track this.